content/blog/2020-08-19-train-a-sequence-model-with-poutyne/index.en.md

```diff
@@ -0,0 +1,563 @@
+bidirectional approach (bi-LSTM). What it means is instead of _simply_ using a seeing the sequence from the start to the end, we
+also train the model to see the sequence from the end to the start. It's important to state that the two directions are
+not shared, meaning that we _see_ the sequence in one direction at the time, but we gather the information from both directions for the 
+fully connected layer. That way, our model can get insight from both directions.
+
+Instead of using only one layer, let's use an LSTM, which means that we use two layers of hidden state.
+
+So, let's create the new LSTM and fully connected network.
+
+```python
+dimension = 300
+num_layer = 2
+bidirectional = True
+
+lstm_network = nn.LSTM(input_size=dimension,
+                       hidden_size=dimension,
+                       num_layers=num_layer,
+                       bidirectional=bidirectional,
+                       batch_first=True)
+
+input_dim = dimension * 2 #since bidirectional
+
+fully_connected_network = nn.Linear(input_dim, tag_dimension)
+
+full_network_bi_lstm = FullNetWork(lstm_network, fully_connected_network)
+```
+ 
+### Training
+ 
+```python
+exp_bi_lstm = Experiment("./", full_network_bi_lstm, device=cuda_device, optimizer=optimizer,
+                 loss_function=cross_entropy, batch_metrics=["acc"])
+exp_bi_lstm.train(train_loader, valid_generator=valid_loader, epochs=epoch_number)
+```
+
+### Results
+Here are the last epoch validation results of the larger model. On the validation dataset,
+we can see that we obtain a marginal gain of around `0.3` for the accuracy over the previous one. Not much of an improvement.
+
+|   Model  | Bi-LSTM two layers |
+|:--------:|:------------------:|
+|   Loss   |    0.0050          |
+| Accuracy |    99.8594         |
+
+But now that we have our two trained models, let's use the test set as a final an **unique** steps of evaluating the performance.
+
+```python
+exp.test(test_loader)
+exp_bi_lstm.test(test_loader)
+```
+
+The next table presents the results of the Bi-LSTM with two layers and the previous model (LSTM with one layer).
+
+|   Model  | LSTM one layer | Bi-LSTM two layers |
+|:--------:|:--------------:|:------------------:|
+|   Loss   |     0.0152     |    **0.0050**      |
+| Accuracy |     99.5758    |    **99.8550**     |
+
+We can see similar results as the validation one for both the model. Also, we still see a little improvement of the accuracy and the loss for the larger model. But when considering that we only improved by around 0.3, one can
+argue that it's only a matter of the seed. To test our approach's robustness, we could use retrain multiple times
+our training step but using a different seed every time. Using those trained models, we can report the mean and one standard variation of the metrics instead of a single training. But instead of doing that, let's try something else. 
```

Comment:
```suggestion
using a different seeds and report the mean and standard deviation of the metrics over all experiments rather than the result of a single training. Let's try something else. 
```